sched/task/task_init.c

```diff
@@ -137,6 +137,10 @@ int nxtask_init(FAR struct task_tcb_s *tcb, const char *name, int priority,
   group->tg_libvars = up_stack_frame(&tcb->cmn, sizeof(struct libvars_s));
   DEBUGASSERT(group->tg_libvars != NULL);
 
+  /* Initialize the task-specific data */
+
+  memset(group->tg_libvars, 0, sizeof(struct libvars_s));
```

Comment:
This clearing of the task-specific data should be performed by ALL architectures.  Hence, I think the memset should be moved into the OS.  Do you agree?  If so, let's close this and I will open another PR the does the memset for all architectures.